types/interceptors.d.ts

```diff
@@ -8,7 +8,7 @@ export default Interceptors
 declare namespace Interceptors {
   export type DumpInterceptorOpts = { maxSize?: number }
   export type RetryInterceptorOpts = RetryHandler.RetryOptions
-  export type RedirectInterceptorOpts = { maxRedirections?: number }
+  export type RedirectInterceptorOpts = { maxRedirections?: number, throwOnMaxRedirections?: boolean }
```

Comment:
Probably worth defining `RedirectHandler.RedirectOptions` like exists for the `RetryHandler`, but that seemed a bit daunting to me [based on all the type definitions for retry-handler.d.ts](https://github.com/nodejs/undici/blob/ca1855491082cb3bacd726afbf7a8e348fc58db4/types/retry-handler.d.ts#L37) so I just left this as a simple one-off type declaration